app/controllers/api/irs_attempts_api_controller.rb

```diff
@@ -54,10 +54,9 @@ def security_event_tokens
     def encrypted_security_event_log_result
       events = security_event_tokens.join("\r\n")
       decoded_key_der = Base64.strict_decode64(IdentityConfig.store.irs_attempt_api_public_key)
-      pub_key = OpenSSL::PKey::RSA.new(decoded_key_der)
 
       IrsAttemptsApi::EnvelopeEncryptor.encrypt(
-        data: events, timestamp: timestamp, public_key: pub_key,
+        data: events, timestamp: timestamp, public_key_str: decoded_key_der,
```

Comment:
Every place that calls `EnvelopeEncryptor#encrypt` is taking the key as a string and having to instantiate an `OpenSSL::PKey::RSA` to pass in. So let's move that into the method rather than doing it repeatedly throughout our code.

Nit: I think I'd like to rename this variable to `public_key`, though this code is probably going away soon.

Reply:
We could also consider pulling in the part that pulls the data from redis based on the timestamp as well, or have two methods:

```
e = EnvelopEncryptor.new(timestamp: timestamp, public_key: public_key)

e.encrypt_from_redis
e.encrypt_events(events)
```